source/common/network/win32_socket_handle_impl.cc

```diff
@@ -0,0 +1,105 @@
+                                                     uint64_t num_slice) {
+  auto result = IoSocketHandleImpl::readv(max_length, slices, num_slice);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::read(Buffer::Instance& buffer,
+                                                    absl::optional<uint64_t> max_length_opt) {
+  auto result = IoSocketHandleImpl::read(buffer, max_length_opt);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::writev(const Buffer::RawSlice* slices,
+                                                      uint64_t num_slice) {
+  auto result = IoSocketHandleImpl::writev(slices, num_slice);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Write);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::write(Buffer::Instance& buffer) {
+  Api::IoCallUint64Result result = IoSocketHandleImpl::write(buffer);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Write);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::sendmsg(const Buffer::RawSlice* slices,
+                                                       uint64_t num_slice, int flags,
+                                                       const Address::Ip* self_ip,
+                                                       const Address::Instance& peer_address) {
+
+  Api::IoCallUint64Result result =
+      IoSocketHandleImpl::sendmsg(slices, num_slice, flags, self_ip, peer_address);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Write);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::recvmsg(Buffer::RawSlice* slices,
+                                                       const uint64_t num_slice, uint32_t self_port,
+                                                       RecvMsgOutput& output) {
+  Api::IoCallUint64Result result =
+      IoSocketHandleImpl::recvmsg(slices, num_slice, self_port, output);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::recvmmsg(RawSliceArrays& slices, uint32_t self_port,
+                                                        RecvMsgOutput& output) {
+  Api::IoCallUint64Result result = IoSocketHandleImpl::recvmmsg(slices, self_port, output);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+Api::IoCallUint64Result Win32SocketHandleImpl::recv(void* buffer, size_t length, int flags) {
+
+  Api::IoCallUint64Result result = IoSocketHandleImpl::recv(buffer, length, flags);
+  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+  return result;
+}
+
+IoHandlePtr Win32SocketHandleImpl::accept(struct sockaddr* addr, socklen_t* addrlen) {
```

Comment:
How about change the `IoSocketHandleImpl::accept` to invoke `SocketInterfaceImpl::makeSocket`, then we needn't implement this accept again.

Reply:
good call, fixed in 87edd58871f6ae965b6a55cd6e443f2b7fc1949a